The sincosf() function used by spirographx if _GNU_SOURCE is defined
is also available on Solaris & OpenSolaris & several other platforms,
not just Linux.

The following patch converts to using a generic autoconf test for the
function to enable its usage.

Submitted upstream at:
https://sourceforge.net/tracker/?func=detail&aid=2767432&group_id=67131&atid=517001

diff -urp -x '*~' -x '*.orig' configure.in configure.in
--- configure.in	2009-04-15 20:14:25.995784000 -0700
+++ configure.in	2009-04-15 20:14:28.005153000 -0700
@@ -7,7 +7,7 @@ AM_CONFIG_HEADER(config.h)
 AC_PROG_AWK
 
 AC_PROG_CC
-AC_GNU_SOURCE
+AC_USE_SYSTEM_EXTENSIONS
 
 if test "x$GCC" = xyes; then
     CFLAGS="-O2"
@@ -39,6 +39,7 @@ PKG_PROG_PKG_CONFIG
 
 # Checks for libraries.
 AC_CHECK_LIB([m], [pow],,AC_MSG_ERROR(-lm was not found but required.))
+AC_CHECK_FUNCS([sincosf])
 AC_ARG_ENABLE(bzip2, [  --disable-bzip2         Don't compress textures/audio with bzip2.])
 if test "x$enable_bzip2" != "xno" ; then
 	AC_CHECK_LIB([bz2], [BZ2_bzBuffToBuffCompress], AC_SUBST(BZIP2, -lbz2) AC_DEFINE([HAVE_LIBBZ2], [], [libbz2 exists.]))
diff -urp -x '*~' -x '*.orig' other_src/spirographx.c other_src/spirographx.c
--- other_src/spirographx.c	2008-08-31 18:36:48.000000000 -0700
+++ other_src/spirographx.c	2009-04-15 20:14:28.006027000 -0700
@@ -94,7 +94,7 @@ void getAll ()
 		const float moverd = (float)m / DETAIL;
 		for (n = 0; n < SUBLOOPS; n++) {
 			const float pointpoweroverdetail = moverd * poweranswer[n];
-#ifdef _GNU_SOURCE
+#ifdef HAVE_SINCOSF
 			float sinppod, cosppod;
 			sincosf(pointpoweroverdetail, &sinppod, &cosppod);
 #else
